### DISCUSSION TITLE
Support for Yubi Keys for Private Key Protection

### DISCUSSION THREAD

**Discussion by SajidPervez** [openpgpjs/openpgpjs#1214] [Ideas]
Hi,
Is there any update regarding support of smartcards like yubi key for private key protection or we still have to rely upon browser native support.

Thanks

**Comment by tomholub** [comment]
I don't think it's actively worked on atm. I suggest it should be implemented as a separate project, so that it can be used directly outside of OpenPGP.js as well. The web is missing a general solution, I think.

**Comment by mahrud** [comment]
From browser this is pretty much impossible until there is native browser support. Node.js, however, can read environment variables and connect to a local socket to talk to a gpg-agent.

However, this feature was asked in #362 but was closed there:
> We're in the process of moving optional functionality to external modules and keeping the core crypto lib nice and lean. I'd recommend a seperate `openpgp-smartcard-storage` npm module for this feature.

**Comment by sundhaug92** [comment]
@mahrud The browser could use WebUSB, but it'd be quite the effort

**Comment by tomholub** [comment]
WebUSB has issues, see comments / conversation at https://github.com/FlowCrypt/flowcrypt-browser/issues/964#issuecomment-461032471
